prow/tide/tide.go

```diff
@@ -170,11 +175,20 @@ func (c *Controller) syncSubpool(ctx context.Context, sp subpool) error {
 	}
 	successes, pendings, nones := accumulate(presubmits, sp.prs, sp.pjs)
 	if len(successes) > 0 {
-		c.log.Infof("Merge PR #%d.", int(pickSmallestNumber(successes).Number))
-	} else if len(pendings) > 0 {
+		if ok, pr := pickSmallestPassingNumber(successes); ok {
+			c.log.Infof("Merge PR #%d.", int(pr.Number))
+			return nil
+		}
+	}
+	if len(pendings) > 0 {
 		c.log.Info("Do nothing. Waiting for pending PRs.")
-	} else if len(nones) > 0 {
-		c.log.Infof("Trigger tests for PR #%d.", int(pickSmallestNumber(nones).Number))
+		return nil
+	}
+	if len(nones) > 0 {
+		if ok, pr := pickSmallestPassingNumber(nones); ok {
+			c.log.Infof("Trigger tests for PR #%d.", int(pr.Number))
```

Comment:
Is this meant to rerun broken tests similar to what the [commenter](https://github.com/kubernetes/test-infra/tree/master/experiment/commenter) is doing?

Reply:
No, it will trigger all jobs on a PR in the tide pool that has passing statuses.

Reply:
Opened https://github.com/kubernetes/test-infra/pull/4270 to discuss more.